packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -1028,7 +1028,16 @@ export const GridMixin = (superClass) =>
       // Check if there is a slotted vaadin-tooltip element.
       const tooltip = this._tooltipController.node;
       if (tooltip && tooltip.isConnected) {
-        this._tooltipController.setTarget(event.target);
+        const target = event.target;
+        const targetRect = target.getBoundingClientRect();
+        const gridRect = this.getBoundingClientRect();
+
+        if (targetRect.left < gridRect.left || targetRect.right > gridRect.right) {
```

Comment:
Should this take `frozen` and `frozen-to-end` columns into account? The cell could be partially behind the cells of a frozen column. Not sure if that's an issue.

Reply:
Good point, thanks. Yes, confirmed that frozen columns can cause this as well. Will also add logic for handling frozen columns (maybe we could just reuse `__isColumnInViewport` helper from `ScrollMixin`).

<img width="309" alt="Screenshot 2024-09-03 at 16 17 15" src="https://github.com/user-attachments/assets/d95dd5e1-964e-4698-8020-d427e9be9ccc">

Reply:
IIRC `__isColumnInViewport` returns `true` even if an element is partially inside viewport

Reply:
Updated the logic to check if the cell is behind `last-frozen` or `first-frozen-to-end` cells and added tests.